File: packages/shared/src/types/report.ts
export type Severity = 'critical' | 'high' | 'medium' | 'low' | 'info'
export type Priority = 'high' | 'medium' | 'low'
export type BugPriority = 'P1' | 'P2' | 'P3' | 'P4'

export interface Finding {
  id: string
  job_id: string
  category: string
  title: string
  description: string
  severity: Severity
  element_hint?: string
  recommendation?: string
  roast_comment?: string
  created_at: string
}

export interface TestStep {
  step: number
  action: string
  expected: string
}

export interface TestCase {
  id: string
  job_id: string
  case_id: string
  title: string
  category: string
  priority: Priority
  preconditions: string[]
  steps: TestStep[]
  expected_result: string
  test_data?: string
  created_at: string
}

export interface BugReport {
  id: string
  job_id: string
  finding_id?: string
  title: string
  severity: Severity
  priority: BugPriority
  steps_to_reproduce: string[]
  expected_result: string
  actual_result: string
  impact: string
  created_at: string
}

export interface Artifact {
  id: string
  job_id: string
  type: 'screenshot' | 'full_page' | 'script' | 'export_md' | 'export_html'
  public_url?: string
  storage_path?: string
  size_bytes?: number
  created_at: string
}

export interface FullReport {
  job: Record<string, unknown>
  findings: Finding[]
  test_cases: TestCase[]
  bug_reports: BugReport[]
  artifacts: Artifact[]
}

File: apps/api/models/report.py
from pydantic import BaseModel
from typing import Optional, List, Literal


class Finding(BaseModel):
    id: str
    category: str
    title: str
    description: str
    severity: Literal["critical", "high", "medium", "low", "info"]
    element_hint: Optional[str] = None
    recommendation: Optional[str] = None
    roast_comment: Optional[str] = None


class AIFindingsResponse(BaseModel):
    findings: List[Finding]
    overall_score: int
    summary: str


class TestStep(BaseModel):
    step: int
    action: str
    expected: str


class TestCase(BaseModel):
    id: str
    case_id: str
    title: str
    category: str
    priority: Literal["high", "medium", "low"]
    preconditions: List[str] = []
    steps: List[TestStep]
    expected_result: str
    test_data: Optional[str] = None


class AITestCasesResponse(BaseModel):
    test_cases: List[TestCase]


class BugReport(BaseModel):
    title: str
    environment: str
    severity: Literal["critical", "high", "medium", "low"]
    priority: Literal["P1", "P2", "P3", "P4"]
    steps_to_reproduce: List[str]
    expected_result: str
    actual_result: str
    impact: str
    suggested_fix: Optional[str] = None
